### ISSUE TITLE
Extraction window: show interactively an extracted definition during its proof

### ISSUE THREAD

**Issue by Zimmi48** [Zimmi48/bugzilla-test#4129]
Note: the issue was created automatically with bugzilla2github tool

# Bugzilla Bug ID: 4129
Date: 2015-03-13 15:40:29 +0100
From: Jonathan Leivent &lt;<jonikelee@gmail.com>&gt;
To:   
CC:   coq-bugs-redist@lists.gforge.inria.fr

Last updated: 2015-03-13 15:40:29 +0100

## Bugzilla Comment ID: 9862
Date: 2015-03-13 15:40:29 +0100
From: Jonathan Leivent &lt;<jonikelee@gmail.com>&gt;

Currently, the restrictions on when one can do extraction on a definition are quite limiting - they include: not until the proof is complete, not within an open Section, and not for an Admitted proof (actually, for this last one, extraction works, but is uninformative).

However, it would be quite useful to be able to watch the impact on extracted code that particular tactics have interactively.  One reason for this is that a tactic may solve a subgoal (or just get closer to solution), but in a way that produces unwanted results in the extracted code.  Another is that the correspondence of subgoals to the unfinished extracted code can help the user keep track of what exactly they are proving in each subgoal.

With such a capability, Coq would be closer to an IDE for programming-by-proving.

From coqdev:

On 03/13/2015 05:24 AM, Pierre Courtieu wrote:
> Hi Jonathan,
>
> I suggest you fill a bug/feature wish about this one. It looks like a
> good idea but will not be implemented for 8.5 I am afraid.
>
> Best regards,
> P.
>
>
> 2015-03-12 19:35 GMT+01:00 Jonathan Leivent <jonikelee@gmail.com>:
>> A small note about the recent change.
>>
>> While in development stages of a complex function via proof mode, I was
>> using admit on some of its unfinished subgoals to make it possible to
>> examine the extracted OCaml of the finished parts. Extraction doesn't work
>> with Admitted or give_up.  So, I added back in a proof_admitted axiom.  No
>> big deal.  But this is another case where admit is useful during
>> development.
>>
>> Or - maybe it would be better to have the ability to see extracted code at
>> any time during an unfinished proof?  Ideally, it would be nice to have
>> another window that shows extracted code for the current definition, which
>> is then updated after every proof step.
>>
>> -- Jonathan
>>
